File: examples/code/home/use-tools/call-tools-with-models/github_chatbot.py
import os
from openai import OpenAI

# Initialize the OpenAI client, pointing to Arcade
client = OpenAI(
    base_url="https://api.arcade.dev/v1",
    api_key="arcade_api_key",
)

# Set a unique identifier for your end user
user_id = "you@example.com"

# Determine which tools will be available to the chatbot agent
tools = [
    "GitHub.SetStarred",
    "Github.ListStargazers",
    "Github.CreateIssue",
    "Github.CreateIssueComment",
    "Github.ListPullRequests",
    "Github.GetPullRequest",
    "Github.UpdatePullRequest",
    "Github.GetRepository",
    "Github.ListRepositoryActivities",
]

while True:
    # Ask the user for input
    prompt = input("Enter your prompt (type 'exit' to quit):")
    if prompt.lower() == "exit":
        break

    # Let the LLM choose a tool to use based on the user's prompt
    response = client.chat.completions.create(
        messages=[
            {"role": "system", "content": "You are a helpful assistant that can interact with GitHub."},
            {"role": "user", "content": prompt},
        ],
        model="gpt-4o",
        user=user_id,
        tools=tools,
        tool_choice="generate",
    )

    print(response.choices[0].message.content)

File: examples/code/home/use-tools/call-tools-with-models/github_chatbot.js
import { OpenAI } from "openai";

// Initialize the OpenAI client, pointing to Arcade
const client = new OpenAI({
  baseURL: "https://api.arcade.dev/v1",
  apiKey: "arcade_api_key",
});

// Set a unique identifier for your end user
let userId = "you@example.com";

// Determine which tools will be available to the chatbot agent
let tools = [
  "GitHub.SetStarred",
  "Github.ListStargazers",
  "Github.CreateIssue",
  "Github.CreateIssueComment",
  "Github.ListPullRequests",
  "Github.GetPullRequest",
  "Github.UpdatePullRequest",
  "Github.GetRepository",
  "Github.ListRepositoryActivities",
];

while (true) {
  // Ask the user for input
  const prompt = await input("Enter your prompt (type 'exit' to quit):");
  if (prompt.toLowerCase() === "exit") {
    break;
  }

  // Use a tool and generate a response
  const response = await client.chat.completions.create({
    messages: [
      {
        role: "system",
        content: "You are a helpful assistant that can interact with GitHub.",
      },
      { role: "user", content: prompt },
    ],
    model: "gpt-4",
    user: userId,
    tools: tools,
    tool_choice: "generate",
  });

  console.log(response.choices[0].message.content);
}
